src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -134,13 +134,14 @@ UserInfoPopup::UserInfoPopup(bool closeAutomatically, QWidget *parent)
     : BaseWindow(closeAutomatically ? userInfoPopupFlagsCloseAutomatically
                                     : userInfoPopupFlags,
                  parent)
+    , closeAutomatically_(closeAutomatically)
     , hack_(new bool)
     , dragTimer_(this)
 {
     this->setWindowTitle("Usercard");
     this->setStayInScreenRect(true);
 
-    if (closeAutomatically)
+    if (this->closeAutomatically_)
         this->setActionOnFocusLoss(BaseWindow::Delete);
     else
         this->setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
Move this section into its own private function, something like `updateFocusLoss` or something which can then be called from the "on pin change" connection

Reply:
I left the case for `this->setAttribute(Qt::WA_DeleteOnClose);` but have both here and the button press call the update focus function.

Reply:
Not a worry any more after merge.